godot-core/src/engine/file.rs

```diff
@@ -0,0 +1,766 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::builtin::{GodotString, PackedByteArray, PackedStringArray, Variant};
+use crate::engine::file_access::{CompressionMode, ModeFlags};
+use crate::gen::central::global::Error;
+use crate::gen::classes::FileAccess;
+use crate::obj::{Gd, Inherits};
+
+use std::cmp;
+use std::io::{BufRead, Error as IoError, ErrorKind, Read, Result as IoResult, Seek, Write};
+
+use super::RefCounted;
+
+/// Open file for reading or writing.
+///
+/// It's a convienience wrapper around [Gd]<[FileAccess]>, providing multiple
+/// both safety and quality of life upgrades over the inner type.
+///
+/// Files by default are alwasy opened with little-endian endianness, as most
+/// files are saved as such. To switch into big_endian_edianness, use
+/// `GFile.set_big_endian()`
+///  
+/// - exposes reading and writing capabilities of [FileAccess] in a safe way,
+/// returning [Result<T>](std::io::Result) instead of the `T` itself
+/// - makes the [FileAccess] exclusive to its instance, not allowing parallel
+/// reads and writes, which could introduce hard to track bugs by unpredicted cursor
+/// movement. Exclusivity also ensures that when the `GFile` moves out of scope,
+/// inner [FileAccess] does also, automatically closing the file. Alternatively,
+/// you can [drop()] the `GFile` to close the file manually.
+/// - implements useful Rust traits allowing better file processing, namely
+/// [Read], [BufRead], [Write], [Seek] and making it better integrated with
+/// various tools in Rust ecosystem (eg. serialization tools).
```

Comment:
Nitpicks about comments:
* Please respect line lengths according to convention (hopefully it's not too much effort to change now, otherwise not so important).
* Start sentences with uppercase letter.
* End sentences with full stop.
* Typo "alwasy"
* "big_endian" -> "big-endian", and no need to repeat "endianess"
* Links referring to API symbols should use backticks, e.g. ```[`Read`]``` instead of `[Read]`
    * You can replace those in the same file with regex search `\[([A-Za-z]+)\]` and replacement ```[`$1`]``` (untested)
    * This might also fix the issue you had with `Gd<T>`

Not just applicable here, but also in other comments.